vars/withCustomCommitStates.groovy

```diff
@@ -0,0 +1,43 @@
+import com.cloudbees.groovy.cps.NonCPS
+
+def call(final String credentialsId, final String sha, final String context, block) {
+
+  def STATE_NAME_PENDING = 'pending'
+  def STATE_NAME_SUCCESS = 'success'
+  def STATE_NAME_FAILURE = 'failure'
+  def STATE_NAME_ERROR = 'error'
+
+  def STATES = [
+    [state: STATE_NAME_PENDING, description: 'The build is pending'],
+    [state: STATE_NAME_SUCCESS, description: 'This commit looks good'],
+    [state: STATE_NAME_FAILURE, description: 'An error or test failure occurred'],
+    [state: STATE_NAME_ERROR, description: 'This commit cannot be built']
+  ]
+
+  def PENDING_STATE = STATES.find{it['state'] == STATE_NAME_PENDING}
+  def SUCCESS_STATE = STATES.find{it['state'] == STATE_NAME_SUCCESS}
+  def FAILURE_STATE = STATES.find{it['state'] == STATE_NAME_FAILURE}
+
+  def currentState = FAILURE_STATE
+  try {
+    setCommitState(credentialsId, sha, PENDING_STATE['state'], PENDING_STATE['description'], context)
+    block()
+    currentState = SUCCESS_STATE
+  } finally {
+    setCommitState(credentialsId, sha, currentState['state'], currentState['description'], context)
+  }
+}
+
+@NonCPS
+setCommitState(final String credentialsId, final String sha, final String state, final String description, final String context) {
+  withCredentials([string(credentialsId: credentialsId, variable: 'OAUTH_TOKEN')]) {
+    sh """
+      curl -XPOST -H "Authorization: token \${OAUTH_TOKEN}" https://api.github.com/repos/h2oai/sparkling-water/statuses/${sha} -d "{
```

Comment:
Repo is hardcoded here. Should be probably argument

Reply:
My bad, sorry, it's still WIP

Reply:
No worries, thanks a lot for this change!